okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -269,29 +269,18 @@ private void startStream(OkHttpClientStream stream, List<Header> requestHeaders)
   private boolean startPendingStreams() {
     boolean hasStreamStarted = false;
     synchronized (lock) {
-      while (!pendingStreams.isEmpty() && streams.size() < maxConcurrentStreams) {
-        PendingStream pendingStream = pendingStreams.poll();
-        startStream(pendingStream.clientStream, pendingStream.requestHeaders);
-        pendingStream.createdFuture.set(null);
-        hasStreamStarted = true;
+      if (!goAway) {
```

Comment:
This goAway check is unnecessary, as pendingStreams is guaranteed to be empty (newStream() checks goAway before adding to pendingStreams). If it is ever non-empty, then that is a separate issue and needs to be fixed, independent of whether we have the check here.